File: source/ResourceLibrary/Controllers/genericController.vb
Option Explicit On
Option Strict On

Imports Contensive.BaseClasses

Namespace Controllers
    Public NotInheritable Class genericController
        Private Sub New()
        End Sub
        '
        '====================================================================================================
        ''' <summary>
        ''' if date is invalid, set to minValue
        ''' </summary>
        ''' <param name="srcDate"></param>
        ''' <returns></returns>
        Public Shared Function encodeMinDate(srcDate As DateTime) As DateTime
            Dim returnDate As DateTime = srcDate
            If srcDate < New DateTime(1900, 1, 1) Then
                returnDate = DateTime.MinValue
            End If
            Return returnDate
        End Function
        '
        '====================================================================================================
        ''' <summary>
        ''' if valid date, return the short date, else return blank string 
        ''' </summary>
        ''' <param name="srcDate"></param>
        ''' <returns></returns>
        Public Shared Function getShortDateString(srcDate As DateTime) As String
            Dim returnString As String = ""
            Dim workingDate As DateTime = encodeMinDate(srcDate)
            If Not isDateEmpty(srcDate) Then
                returnString = workingDate.ToShortDateString()
            End If
            Return returnString
        End Function
        '
        '====================================================================================================
        Public Shared Function isDateEmpty(srcDate As DateTime) As Boolean
            Return (srcDate < New DateTime(1900, 1, 1))
        End Function
        '
        '====================================================================================================
        Public Shared Function getSortOrderFromInteger(id As Integer) As String
            Return id.ToString().PadLeft(7, "0"c)
        End Function
        '
        '====================================================================================================
        Public Shared Function getDateForHtmlInput(source As DateTime) As String
            If isDateEmpty(source) Then
                Return ""
            Else
                Return source.Year.ToString() + "-" + source.Month.ToString().PadLeft(2, "0"c) + "-" + source.Day.ToString().PadLeft(2, "0"c)
            End If
        End Function
        '
        '====================================================================================================
        Public Shared Function convertToDosPath(sourcePath As String) As String
            Return sourcePath.Replace("/", "\")
        End Function
        '
        '====================================================================================================
        Public Shared Function convertToUnixPath(sourcePath As String) As String
            Return sourcePath.Replace("\", "/")
        End Function
        '
        Public Shared Function Main_getAddonOption(cp As CPBaseClass, requestName As String, ignore As String) As String
            Return cp.Doc.GetText(requestName)
        End Function
        '
        Public Shared Function getBoolean_Main_getAddonOption(cp As CPBaseClass, requestName As String, ignore As String) As Boolean
            Return cp.Doc.GetBoolean(requestName)
        End Function
        '
        Public Shared Sub Main_testpoint(cp As CPBaseClass, message As String)
            cp.Site.TestPoint(message)
        End Sub
        '
        Public Shared Function KmaEncodeSQLNumber(cp As CPBaseClass, src As Integer) As String
            Return cp.Db.EncodeSQLNumber(src)
        End Function
        '
        Public Shared Function KmaEncodeSQLNumber(cp As CPBaseClass, src As Double) As String
            Return cp.Db.EncodeSQLNumber(src)
        End Function
        Public Shared Function KmaEncodeSQLText(cp As CPBaseClass, src As String) As String
            Return cp.Db.EncodeSQLText(src)
        End Function
        '
        Public Shared Function htmlDiv(innerHtml As String, Optional htmlClass As String = "", Optional htmlId As String = "") As String
            Dim result As String = "<div "
            result += If(String.IsNullOrEmpty(htmlClass), "", " class=""" & htmlClass & """")
            result += If(String.IsNullOrEmpty(htmlId), "", " id=""" & htmlId & """")
            Return result & ">" & innerHtml & "</div>"
        End Function
        '
        Public Shared Function htmlButton(value As String, Optional htmlClass As String = "", Optional htmlId As String = "", Optional onClick As String = "") As String
            Dim result As String = "<button name=""name"" value=""" & value & """"
            result += If(String.IsNullOrEmpty(htmlClass), "", " class=""" & htmlClass & """")
            result += If(String.IsNullOrEmpty(htmlId), "", " id=""" & htmlId & """")
            result += If(String.IsNullOrEmpty(onClick), "", " onClick=""" & onClick & """")
            Return result & ">"
        End Function
        '
        Public Shared Function htmlHidden(htmlName As String, htmlValue As String, Optional htmlClass As String = "", Optional htmlId As String = "") As String
            Dim result As String = "<input type=hidden name=""" & htmlName & """ value=""" & htmlValue & """"
            result += If(String.IsNullOrEmpty(htmlClass), "", " class=""" & htmlClass & """")
            result += If(String.IsNullOrEmpty(htmlId), "", " id=""" & htmlId & """")
            Return result & ">"
        End Function
        '
        Public Shared Function htmlHidden(htmlName As String, htmlValue As Integer, Optional htmlClass As String = "", Optional htmlId As String = "") As String
            Return htmlHidden(htmlName, htmlValue.ToString(), htmlClass, htmlId)
        End Function
        '
        Public Shared Function adminUrl(cp As CPBaseClass) As String
            Return cp.Site.GetText("adminurl")
        End Function
        '
        Public Shared Function kmaEncodeURL(cp As CPBaseClass, url As String) As String
            Return cp.Utils.EncodeUrl(url)
        End Function
        '
        Public Shared Function Main_GetPanel(src As String) As String
            Return htmlDiv(src)
        End Function
        '
        Public Shared Function kmaEncodeJavascript(cp As CPBaseClass, src As String) As String
            Return cp.Utils.EncodeJavascript(src)
        End Function
        '
        Public Shared Function kmaAddSpan(innerHtml As String, htmlClass As String) As String
            Return "<span class=""" & htmlClass & """>" & innerHtml & "</span>"
        End Function
        '
        Public Shared Function Main_GetFormInputCheckBox(htmlName As String, isChecked As Boolean) As String
            Return "<input type=checkbox name=""" & htmlName & """ value=1 " & If(isChecked, " checked", "") & ">"
        End Function
        '
        Public Shared Function Main_GetPanelInput(innerHtml As String) As String
            Return htmlDiv(innerHtml)
        End Function
        '
        Public Shared Function KmaEncodeMissingText(src As String, Optional ignore As String = "") As String
            Return src
        End Function
        '
        Public Shared Function KmaEncodeMissingBoolean(src As Boolean, Optional ignore As Boolean = False) As Boolean
            Return src
        End Function
        '
        Public Shared Function KmaEncodeMissingBoolean(src As Boolean, Optional ignore As String = "") As Boolean
            Return src
        End Function
        '
        Public Shared Function kmaEncodeHTML(cp As CPBaseClass, src As String) As String
            Return cp.Utils.EncodeHTML(src)
        End Function
        '

    End Class
End Namespace

File: source/ResourceLibrary/Controllers/genericController.cs

using System;
using Contensive.BaseClasses;

namespace Contensive.Addons.ResourceLibrary.Controllers {
    public sealed class genericController {
        private genericController() {
        }
        public static DateTime encodeMinDate(DateTime srcDate) {
            DateTime returnDate = srcDate;
            if (srcDate < new DateTime(1900, 1, 1)) {
                returnDate = DateTime.MinValue;
            }
            return returnDate;
        }
        public static string getShortDateString(DateTime srcDate) {
            string returnString = "";
            DateTime workingDate = encodeMinDate(srcDate);
            if (!isDateEmpty(srcDate)) {
                returnString = workingDate.ToShortDateString();
            }
            return returnString;
        }
        public static bool isDateEmpty(DateTime srcDate) {
            return (srcDate < new DateTime(1900, 1, 1));
        }
        public static string getSortOrderFromInteger(int id) {
            return id.ToString().PadLeft(7, '0');
        }
        public static string getDateForHtmlInput(DateTime source) {
            if (isDateEmpty(source)) {
                return "";
            } else {
                return $"{source.Year}-{source.Month.ToString().PadLeft(2, '0')}-{source.Day.ToString().PadLeft(2, '0')}";
            }
        }
        public static string convertToDosPath(string sourcePath) {
            return sourcePath.Replace("/", "\\");
        }
        public static string convertToUnixPath(string sourcePath) {
            return sourcePath.Replace("\\", "/");
        }
        public static string Main_getAddonOption(CPBaseClass cp, string requestName, string ignore) {
            return cp.Doc.GetText(requestName);
        }
        public static bool getBoolean_Main_getAddonOption(CPBaseClass cp, string requestName, string ignore) {
            return cp.Doc.GetBoolean(requestName);
        }
        public static void Main_testpoint(CPBaseClass cp, string message) {
            cp.Site.TestPoint(message);
        }
        public static string KmaEncodeSQLNumber(CPBaseClass cp, int src) {
            return cp.Db.EncodeSQLNumber(src);
        }
        public static string KmaEncodeSQLNumber(CPBaseClass cp, double src) {
            return cp.Db.EncodeSQLNumber(src);
        }
        public static string KmaEncodeSQLText(CPBaseClass cp, string src) {
            return cp.Db.EncodeSQLText(src);
        }
        public static string htmlDiv(string innerHtml, string htmlClass = "", string htmlId = "") {
            string result = "<div ";
            result += string.IsNullOrEmpty(htmlClass) ? "" : $" class=\"{htmlClass}\"";
            result += string.IsNullOrEmpty(htmlId) ? "" : $" id=\"{htmlId}\"";
            return $"{result}>{innerHtml}</div>";
        }
        public static string htmlButton(string value, string htmlClass = "", string htmlId = "", string onClick = "") {
            string result = $"<button name=\"name\" value=\"{value}\"";
            result += string.IsNullOrEmpty(htmlClass) ? "" : $" class=\"{htmlClass}\"";
            result += string.IsNullOrEmpty(htmlId) ? "" : $" id=\"{htmlId}\"";
            result += string.IsNullOrEmpty(onClick) ? "" : $" onClick=\"{onClick}\"";
            return $"{result}>";
        }
        public static string htmlHidden(string htmlName, string htmlValue, string htmlClass = "", string htmlId = "") {
            string result = $"<input type=hidden name=\"{htmlName}\" value=\"{htmlValue}\"";
            result += string.IsNullOrEmpty(htmlClass) ? "" : $" class=\"{htmlClass}\"";
            result += string.IsNullOrEmpty(htmlId) ? "" : $" id=\"{htmlId}\"";
            return $"{result}>";
        }
        public static string htmlHidden(string htmlName, int htmlValue, string htmlClass = "", string htmlId = "") {
            return htmlHidden(htmlName, htmlValue.ToString(), htmlClass, htmlId);
        }
        public static string adminUrl(CPBaseClass cp) {
            return cp.Site.GetText("adminurl");
        }
        public static string kmaEncodeURL(CPBaseClass cp, string url) {
            return cp.Utils.EncodeUrl(url);
        }
        public static string Main_GetPanel(string src) {
            return htmlDiv(src);
        }
        public static string kmaEncodeJavascript(CPBaseClass cp, string src) {
            return cp.Utils.EncodeJavascript(src);
        }
        public static string kmaAddSpan(string innerHtml, string htmlClass) {
            return $"<span class=\"{htmlClass}\">{innerHtml}</span>";
        }
        public static string Main_GetFormInputCheckBox(string htmlName, bool isChecked) {
            return $"<input type=checkbox name=\"{htmlName}\" value=1 {(isChecked ? " checked" : "")}>";
        }
        public static string Main_GetPanelInput(string innerHtml) {
            return htmlDiv(innerHtml);
        }
        public static string KmaEncodeMissingText(string src, string ignore = "") {
            return src;
        }
        public static bool KmaEncodeMissingBoolean(bool src, bool ignore = false) {
            return src;
        }
        public static bool KmaEncodeMissingBoolean(bool src, string ignore = "") {
            return src;
        }
        public static string kmaEncodeHTML(CPBaseClass cp, string src) {
            return cp.Utils.EncodeHTML(src);
        }
    }
}
